lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -175,6 +181,41 @@ def set_pki_intermediate_ca_certificate(self, certificate: str, mount: str) -> N
         )
         logger.info("Set the intermediate CA certificate for the PKI backend")
 
+    def sign_pki_certificate_signing_request(
+        self,
+        mount: str,
+        role: str,
+        csr: str,
+        common_name: str,
+    ) -> Certificate:
+        """Sign a certificate signing request for the PKI backend.
+
+        Args:
+            mount: The PKI mount point.
+            role: The role to use for signing the certificate.
+            csr: The certificate signing request.
+            common_name: The common name for the certificate.
+
+        Returns:
+            Certificate: The signed certificate object
+        """
+        try:
+            response = self._client.secrets.pki.sign_certificate(
+                csr=csr,
+                mount_point=mount,
+                common_name=common_name,
+                name=role,
+            )
+            logger.info("Signed a PKI certificate for %s", common_name)
+            return Certificate(
+                certificate=response["data"]["certificate"],
+                ca=response["data"]["issuing_ca"],
+                chain=response["data"]["ca_chain"],
+            )
+        except InvalidRequest as e:
+            logger.warning("Error while signing PKI certificate: %s", e)
+            raise VaultClientError("Error while signing PKI certificate: %s") from e
```

Comment:
Do we actually want vault to crash if we fail to sign a certificate? What if the CSR is malformed? A single requirer crashing the whole charm?

Reply:
The charm catches `VaultClientErrors` meaning this won't crash the whole charm

Reply:
Done, we now return None here.